queue/direct.go

```diff
@@ -639,23 +639,32 @@ func deliverHost(log mlog.Log, resolver dns.Resolver, dialer smtpclient.Dialer,
 			err = fmt.Errorf("storing recipient domain tls status: %w", err)
 		}
 	}
-	if err != nil {
+
+	inspectError := func(err error) error {
+		if err == nil {
```

Comment:
this can't happen right? seems like only calling it for actual errors is a good invariant, so we can get rid of the check.